atlasdb-client/src/main/java/com/palantir/atlasdb/stream/BlockConsumingInputStream.java

```diff
@@ -22,6 +22,7 @@
 import com.google.common.base.Preconditions;
 
 public final class BlockConsumingInputStream extends InputStream {
+    private static final int MAX_ARRAY_SIZE = Integer.MAX_VALUE - 8; // ArrayList.MAX_ARRAY_SIZE on 64-bit systems
```

Comment:
If we're going to do this, Atlas should expose this constant somewhere so that internal products can use it.  Otherwise internal products still attempt to use Integer.MAX_VALUE.